dm/master/election_test.go

```diff
@@ -99,4 +99,6 @@ func (t *testElectionSuite) TestFailToStartLeader(c *check.C) {
 	_, leaderID, _, err = s2.election.LeaderInfo(ctx)
 	c.Assert(err, check.IsNil)
 	c.Assert(leaderID, check.Equals, cfg2.Name)
+
+	cancel()
```

Comment:
This cancal is redundant, already called `defer cancel()` 🤔

Reply:
Actually no, the current test case relies on `ctx.cancel()` to exit and if ctx doesn't cancel, the test will never end (dm-master's server is keep  running in backgroud)🤣

Reply:
means upper defer stack relies on context cancel